src/FieldRenameHelper.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Drupal\localgov_core;
+
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\Core\Config\FileStorage;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+
+class FieldRenameHelper {
+
+  /**
+   * Rename field
+   * @param  string $old_field_name
+   *   Old field machine name.
+   * @param  string $new_field_name
+   *   New field machine name.
+   * @param  string $entity_type
+   *   Entity type name (eg. node, paragraph etc...)
+   */
+  public static function renameField(string $old_field_name, string $new_field_name, string $entity_type) {
+
+    // Rename field.
+    // https://www.drupal.org/docs/drupal-apis/update-api/updating-entities-and-fields-in-drupal-8
+
+    // Make sure new field name is correctly formatted.
+    $new_field_name = substr($new_field_name, 0, 32);
+
+    // Field variables.
+    // @TODO find out how to get the correct table names.
+    $old_table = $entity_type . '__' . $old_field_name;
+    $new_table = $entity_type . '__' . $new_field_name;
+    $revision_old_table = $entity_type . '_revision__' . $old_field_name;
+    $revision_new_table = $entity_type . '_revision__' . $new_field_name;
```

Comment:
This is only relevant when ```$field_storage->isRevisionalbe()``` applies.  But probably won't do us much harm, so can be left intact :)

Reply:
For the moment these are nodes, however would be useful in future.